packages/react-compose/src/mergeProps.ts

```diff
@@ -7,10 +7,13 @@ import { resolveSlotProps } from './resolveSlotProps';
  * It's expected that the component will call mergeProps(state, options) from within
  * render; after resolving state and before rendering slots and slotProps.
  */
-export function mergeProps<TProps, TState = TProps>(
-  state: TState,
-  options: ComposePreparedOptions<TProps, TState>,
-): MergePropsResult<TState> {
+export function mergeProps<
+  TProps,
+  TState = TProps,
+  TSlots = GenericDictionary,
+  // tslint:disable-next-line:no-any
+  TSlotProps = { [key in keyof TSlots]: any }
+>(state: TState, options: ComposePreparedOptions<TProps, TState>): MergePropsResult<TState, TSlots, TSlotProps> {
   const result: MergePropsResult<TState> = {
```

Comment:
You could do:
```suggestion
  const result: MergePropsResult<TState, TSlots, TSlotProps> = {
```

so that you don't have to do the `as` when you return.

Reply:
if you look at `result` value, it's not type `MergePropsResult<TState, TSlots, TSlotProps>`. only after resolving, it becomes `MergePropsResult<TState, TSlots, TSlotProps>`, hence i'm casting it later